### DISCUSSION TITLE
Exit from app on pop in last child

### DISCUSSION THREAD

**Discussion by Muhammadali143904** [arkivanov/Decompose#183] [Q&A] [answered]
Hi, I have a question, what we need call(function not back button) to exit from app, pop isn't exit
Thank you beforehand

**Comment by arkivanov** [comment, marked as answer]
Hello, what kind of application? Android, JVM desktop, iOS? How do you pop the stack?

**Reply by Muhammadali143904** [reply]
Android

In Home screen i have a Outputs, one of them is Exit(because i can exit from button), i am calling a navigation.pop(), but it isn't exit, and in RootComponent i can't call requireActivity().finish()

**Reply by arkivanov** [reply]
Yes, you can't pop the last component from the stack. You should call `pop` and check the result, if it's `false` then close the app manually.

```
class HomeComponent(
    componentContext: ComponentContext,
    private val onFinished: () -> Unit,
) : ComponentContext by componentContext {
    private val navigation = StackNavigation<...>()
    // ...
    
    private fun exit() {
        navigation.pop { result ->
            if (!result) {
                onFinished()
            }
        }
    }
}
```

**Reply by Muhammadali143904** [reply]
Oh, Thank you
